com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/helpers/ApplicationColorDetector.java

```diff
@@ -1,29 +1,52 @@
 package com.sap.cloud.lm.sl.cf.core.helpers;
 
+import java.util.Arrays;
 import java.util.Date;
+import java.util.List;
 
+import org.flowable.variable.api.history.HistoricVariableInstance;
+
+import com.sap.cloud.lm.sl.cf.core.Constants;
+import com.sap.cloud.lm.sl.cf.core.dao.OperationDao;
+import com.sap.cloud.lm.sl.cf.core.flowable.FlowableFacade;
 import com.sap.cloud.lm.sl.cf.core.message.Messages;
 import com.sap.cloud.lm.sl.cf.core.model.ApplicationColor;
 import com.sap.cloud.lm.sl.cf.core.model.DeployedMta;
 import com.sap.cloud.lm.sl.cf.core.model.DeployedMtaModule;
+import com.sap.cloud.lm.sl.cf.core.model.Phase;
+import com.sap.cloud.lm.sl.cf.web.api.model.Operation;
+import com.sap.cloud.lm.sl.cf.web.api.model.State;
 import com.sap.cloud.lm.sl.common.ConflictException;
 
 public class ApplicationColorDetector {
 
     private static final ApplicationColor COLOR_OF_APPLICATIONS_WITHOUT_SUFFIX = ApplicationColor.BLUE;
 
-    public ApplicationColor detectFirstDeployedApplicationColor(DeployedMta deployedMta) {
+    public ApplicationColor detectLiveApplicationColor(DeployedMta deployedMta, OperationDao operationDao, String correlationId,
```

Comment:
Annotate the class with `@Named` (thus making it a Spring component) and inject the `operationDao` and `flowableFacade` through Spring.